twilight-model/src/channel/message/reaction_type.rs

```diff
@@ -0,0 +1,65 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Copy, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+#[non_exhaustive]
+#[serde(from = "u8", into = "u8")]
+/// The kind of reaction.
+pub enum ReactionType {
+    /// A non-burst/super reaction.
+    Normal,
+    /// A super reaction.
+    Burst,
+    /// An unknown reaction type.
+    Unknown(u8),
+}
+
+impl From<u8> for ReactionType {
+    fn from(value: u8) -> Self {
+        match value {
+            0 => Self::Normal,
+            1 => Self::Burst,
+            unknown => Self::Unknown(unknown),
+        }
+    }
+}
+
+impl From<ReactionType> for u8 {
+    fn from(value: ReactionType) -> Self {
+        match value {
+            ReactionType::Normal => 0,
+            ReactionType::Burst => 1,
+            ReactionType::Unknown(unknown) => unknown,
+        }
+    }
+}
+
+impl ReactionType {
+    /// The name of the reaction type.
+    pub const fn name(self) -> &'static str {
+        match self {
+            Self::Normal => "Normal",
+            Self::Burst => "Burst",
```

Comment:
The docs use ALL CAPS for the names so we should do that here as well.
```suggestion
            Self::Normal => "NORMAL",
            Self::Burst => "BURST",
```

Reply:
iirc this isnt what we do for other enums implementing a similar method

Reply:
As said elsewhere, let us skip this for now.